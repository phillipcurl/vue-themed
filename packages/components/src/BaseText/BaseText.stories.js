import { action } from "@storybook/addon-actions";
import { linkTo } from "@storybook/addon-links";
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";

export default {
  title: "BaseText",
  decorators: [withKnobs, withA11y]
};

export const Default = () => ({
  props: {
    text: {
      default: text(
        "Text",
        "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dignissimos dicta itaque hic! Perspiciatis in harum aliquam maiores quia dicta, culpa nam? Voluptas inventore incidunt ut? Sapiente sint tenetur placeat officiis?"
      )
    }
  },
  template: "<BaseText>{{text}}</BaseText>"
});
Default.story = {
  name: "BaseText",
  parameters: {
    notes:
      "The default button component with no additional styles or variants set."
  }
};
