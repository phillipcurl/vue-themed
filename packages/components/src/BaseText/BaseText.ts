import Vue from "vue";

export default Vue.extend({
  name: "BaseText",
  props: {
    maxWidth: {
      type: [String, Array],
      default: "66ch"
    },
    lineHeight: {
      type: String,
      default: "body"
    },
    as: {
      type: String,
      default: "p"
    },
    css: {
      type: Object,
      default: () => ({})
    }
  },
  render: function(h) {
    return h(
      this.as,
      {
        // class: this.$theme.css(
        //   Object.assign({}, this.css, {
        //     maxWidth: this.maxWidth,
        //     margin: "0px",
        //     lineHeight: this.lineHeight,
        //     color: "text"
        //   })
        // ),
        attrs: this.$attrs
      },
      this.$slots.default
    );
  }
});
