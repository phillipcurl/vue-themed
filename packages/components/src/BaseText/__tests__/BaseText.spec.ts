import { shallowMount } from "@vue/test-utils";

import BaseText from "../BaseText";
import Theme from "./../../../plugin/theme";

const $theme = new Theme();

describe("BaseText", () => {
  test("is a Vue instance", () => {
    const wrapper = shallowMount(BaseText, {
      mocks: {
        $theme
      }
    });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("is a Vue instance", () => {
    const wrapper = shallowMount(BaseText, {
      mocks: {
        $theme
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
