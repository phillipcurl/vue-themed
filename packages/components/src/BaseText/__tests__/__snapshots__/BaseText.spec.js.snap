// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText is a Vue instance 1`] = `
<p
  class="css-15maqve"
/>
`;
